include/GafferVDB/VDBObject.h

```diff
@@ -0,0 +1,97 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2015, John Haddon. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef GAFFERVDB_VDBOBJECT_H
+#define GAFFERVDB_VDBOBJECT_H
+
+
+#include "ImathBox.h"
+#include "openvdb/openvdb.h"
+
+#include "IECore/Object.h"
+#include "IECore/CompoundObject.h"
+#include "IECore/VisibleRenderable.h"
+#include "IECore/VectorTypedData.h"
+
+#include "GafferVDB/TypeIds.h"
+
+namespace GafferVDB
+{
+
+class VDBObject : public IECore::VisibleRenderable
+{
+
+	public :
+
+		VDBObject();
+
+		//! initialise VDBObject from a vdb file
+		VDBObject(const std::string& fileanme);
+
+		IE_CORE_DECLAREEXTENSIONOBJECT( GafferVDB::VDBObject, VDBObjectTypeId, IECore::VisibleRenderable );
+
+		openvdb::GridBase::ConstPtr findGrid(const std::string& name) const;
+		openvdb::GridBase::Ptr findGrid(const std::string& name);
+		void removeGrid(const std::string& name);
+
+		std::vector<std::string> gridNames() const;
+
+		void addGrid(openvdb::GridBase::Ptr grid);
+
+		Imath::Box3f bound() const override;
+		void render( IECore::Renderer *renderer ) const override;
+
+		IECore::UCharVectorDataPtr memoryBuffer() const;
+
+		void forceRead(const std::string& name);
```

Comment:
Bump

Reply:
This is only used in a poor (not very good) test to see how much grid data we're loading. It can be removed or perhaps the name changed to 'ensureResident'? There isn't a good case for keeping it in the API but it's handy in the tests to demonstrate this aspect of VDB.

Reply:
Thanks for the explanation. Ideally I think VDBObject wouldn't have either `forceRead()` or `memoryBuffer()`, so it stayed closer to the principle of "a lightweight container of grids", and left you to use the VDB API directly. I wonder if it's possible to wrap VDBObject such that we can get access to the grids in Python for use in the tests and elsewhere? From a quick look at the VDB bindings, it looks as if perhaps we could do that quite easily by borrowing [this function](https://github.com/dreamworksanimation/openvdb/blob/master/openvdb/python/pyGrid.h#L100)? If that turns out to be non-trivial then don't worry about it, but it might be worth a look?